.github/workflows/build-nginx-proxy-image.yml

```diff
@@ -0,0 +1,174 @@
+name: Build Nginx Proxy Docker Image
+
+env:
+  IMAGE_NAME: nginx-proxy
+  AWS_REGION: eu-west-1
+
+on:
+  pull_request:
+  push:
+    branches:
+      - master
+
+jobs:
+  test:
+    name: Test Pay Nginx Proxy
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout Repo
+        uses: actions/checkout@a12a3943b4bdde767164f792f33f40b04645d846
+        with:
+          fetch-depth: '1'
+
+      - name: Run Shell Check on go.sh
+        shell: bash
+        run: docker run --rm -v "${GITHUB_WORKSPACE}:/mnt" koalaman/shellcheck:stable@sha256:ec26f2ad1492bf4343a7f82227fec67a19e4729b5ad02f6a2fe1a2b44cc037ff -x go.sh
+
+      - name: Build Docker Compose Testing Stack
+        shell: bash
+        run: |
+          cd ${GITHUB_WORKSPACE}/tests
```

Comment:
Rather than cding in you can set the working_directory which means you don't need the GITHUB_WORKSPACE bonanza :D

```
- name: ... 
    working-directory: tests
    run: docker-compose build
```